### PR TITLE
ci: add coverage

### PR DESCRIPTION
**Pull request by hugocaillard**

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+coverage:
+  status:
+    project:
+      default:
+        # threshold: 0.005  # 0.5 %
+        threshold: 0.10 # 10 %
```
Comment (hugocaillard) [inline codecov.yml:6]: huge threshold for now. Just to allow merging this PR without bypassing protection. Will decrease later

**Comment by hugocaillard** [comment]
@CharlieC3 
I added back the the codecov report on the "old" ci.
Unfortunately, we are still not using the new, faster version (which should be even faster now that we removed Deno).

But now that this repo is in a more stable state (post clarinet v2 release), it'll make more sense to work on a clean CI and workflow

File: codecov.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+coverage:
+  status:
+    project:
+      default:
+        # threshold: 0.005  # 0.5 %
+        threshold: 0.15 # 15 %
```
Comment (hugocaillard) [inline codecov.yml:6]: huge threshold for now. Just to allow merging this PR without bypassing protection. Will decrease later

File: codecov.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+coverage:
+  status:
+    patch: false
+    project:
+      default:
+        # threshold: 0.005  # 0.5 %
+        threshold: 15%
```
Comment (hugocaillard) [inline codecov.yml:7]: Huge threshold for now. Just to allow merging this PR without bypassing protection. Will decrease later

**Review by CharlieC3 — APPROVED** [review]
Seems fine to me! :shipit:  I don't necessarily like that we have to duplicate more workflow logic. I'm not suggesting we do anything about it right now, but eventually I would like this project to migrate to the "component" workflows.

**Comment by hugocaillard** [comment]
Thanks @CharlieC3 
Indeed, switching to the "new" CI should be a Q4 focus

**Pull request merged**